### PR TITLE
Caching the OAuth tokens in Application Gateway

### PR DESCRIPTION
**Pull request by mvshao**
**Description**

Changes proposed in this pull request:

- OAuth tokens are stored per whole OAuth credentials -> `clientID + clientSecret + authURL`
- mTLS OAuth tokens are stored per whole credentials -> `clientID + sha256Certificate + sha256Key + authURL`
- minor improvement in the documentation regarding Application Gateway testing

**Related issue(s)**
For more information visit https://extra.secret.repository/kyma/backlog/issues/2774

### CODE REVIEW

**Review by majakurcius — APPROVED** [review]

File: tests/components/application-connector/resources/charts/gateway-test/charts/test/templates/applications/credentials/mtls-oauth-nagative-other-ca.yaml [hunk 1]
```diff
@@ -6,6 +6,6 @@ metadata:
 type: Opaque
 data:
   {{- $files := .Files }}
-  crt: {{ $files.Get "certs/negative/client.crt" | b64enc  }}
-  key: {{ $files.Get "certs/negative/client.key" | b64enc  }}
-  clientId: {{ "someClientID1" | b64enc }}
+  crt: {{ $files.Get "certs/invalid-ca/client.crt" | b64enc  }}
+  key: {{ $files.Get "certs/invalid-ca/client.key" | b64enc  }}
+  clientId: {{ "clientID" | b64enc }}
```
Comment (franpog859) [inline tests/components/application-connector/resources/charts/gateway-test/charts/test/templates/applications/credentials/mtls-oauth-nagative-other-ca.yaml:11]: I think in the tests, we can change all the clientIDs from `someClientID...` to just `clientID`, right?

File: tests/components/application-connector/resources/charts/gateway-test/charts/test/templates/applications/credentials/mtls-oauth-nagative-other-ca.yaml [hunk 2]
```diff
@@ -6,6 +6,6 @@ metadata:
 type: Opaque
 data:
   {{- $files := .Files }}
-  crt: {{ $files.Get "certs/negative/client.crt" | b64enc  }}
-  key: {{ $files.Get "certs/negative/client.key" | b64enc  }}
-  clientId: {{ "someClientID1" | b64enc }}
+  crt: {{ $files.Get "certs/invalid-ca/client.crt" | b64enc  }}
+  key: {{ $files.Get "certs/invalid-ca/client.key" | b64enc  }}
```
Comment (franpog859) [inline tests/components/application-connector/resources/charts/gateway-test/charts/test/templates/applications/credentials/mtls-oauth-nagative-other-ca.yaml:10]: Why is the `...negative...` changed to the `...invalid-ca...` there?

**Review by franpog859 — COMMENTED** [review]
Please, add also the test case for the normal OAuth, mentioned in the issue

File: tests/components/application-connector/resources/charts/gateway-test/charts/test/templates/applications/credentials/mtls-oauth-nagative-other-ca.yaml (hunk 2 above)
Comment (mvshao) [reply tests/components/application-connector/resources/charts/gateway-test/charts/test/templates/applications/credentials/mtls-oauth-nagative-other-ca.yaml:10]: The same CA signs a positive certificate and a negative certificate. For this test, I generated certificates that are signed by other CA.

File: tests/components/application-connector/resources/charts/gateway-test/charts/test/templates/applications/credentials/mtls-oauth-nagative-other-ca.yaml (hunk 1 above)
Comment (mvshao) [reply tests/components/application-connector/resources/charts/gateway-test/charts/test/templates/applications/credentials/mtls-oauth-nagative-other-ca.yaml:11]: From now on, the `clientId` field is not used to create the mTLS cache key. We can set it to any value in the corresponding tests.

Comment (mvshao) [reply tests/components/application-connector/resources/charts/gateway-test/charts/test/templates/applications/credentials/mtls-oauth-nagative-other-ca.yaml:11]: Yes, we can also change that

File: tests/components/application-connector/resources/charts/gateway-test/charts/test/templates/applications/credentials/oauth-negative-incorrect-id.yaml [hunk 3]
```diff
@@ -5,5 +5,5 @@ metadata:
   namespace: kyma-integration
 type: Opaque
 data:
-  clientId: {{ "bad id" | b64enc }}
+  clientId: {{ "clientID" | b64enc }}
```
Comment (franpog859) [inline tests/components/application-connector/resources/charts/gateway-test/charts/test/templates/applications/credentials/oauth-negative-incorrect-id.yaml:8]: I believe this one should not be changed :d It's the `oauth-negative-incorrect-id` case

**Review by franpog859 — APPROVED** [review]

**Review by grego952 — APPROVED** [review]

**Pull request merged**